homeassistant/components/flume/coordinator.py

```diff
@@ -33,3 +42,74 @@ async def _async_update_data(self) -> None:
             self.flume_device.values,
             self.flume_device.query_payload,
         )
+
+
+class FlumeNotificationDataUpdateCoordinator(DataUpdateCoordinator[None]):
+    """Data update coordinator for flume notifications."""
+
+    def __init__(self, hass: HomeAssistant, auth) -> None:
+        """Initialize the Coordinator."""
+        super().__init__(
+            hass,
+            name=DOMAIN,
+            logger=_LOGGER,
+            update_interval=NOTIFICATION_SCAN_INTERVAL,
+        )
+        self.auth = auth
+        self.notifications: dict = {}
+        self.active_notifications_by_device: dict = {}
+
+    def _update_lists(self):
+        """Query flume for notification list."""
+        self.notifications = pyflume.FlumeNotificationList(
+            self.auth, read="true"
+        ).notification_list
+        _LOGGER.debug("Notifications %s", self.notifications)
+
+        # The list of notifications returned by API will be in chronological order and there may be
+        # multiple notifications of the same type (such as High Flow). The only special
+        # case here is bridge notifications: NOTIFICATION_BRIDGE_DISCONNECT, which will be either True
+        # or False as indicated in the ["extra"][BRIDGE_NOTIFICATION_KEY].
+        # full_notification_info = {}
+
+        notifications_by_device: dict[str, set[str]] = {}
+
+        for item in self.notifications:
+
+            device_id = item["device_id"]
+            rule = item["extra"]["event_rule_name"]
+
+            # Because bridge is a disconnect notification, True = disconnected - its value needs to be reversed
+            # all other notifications will be not False ... aka True
+            value = not item["extra"].get(BRIDGE_NOTIFICATION_KEY, False)
+
+            if value:
+                notifications_by_device.setdefault(
+                    device_id, {BRIDGE_NOTIFICATION_RULE}
```

Comment:
Looking much better

Can we avoid adding it and then discarding it ?

Reply:
The issues is we get an array of notifications and if the user hasn't cleared them in the past we could get 

BRIDGE DISCONNNECT
BRIDGE CONNECT
BRIDGE DISCONNECT

I guess if need be I could filter the entire set of notification on BRIDGE_NOTIFICATION and take the last one.... and if its disconnect add it, and if its not don't... 

I'll try to mock that up and you can take a look.

But I'm not sure if that's more or less elegant than removing a notification...

Reply:
I think I may have messed this up actually... because I'm not looking at device id...

Reply:
Ok rewrote the code a bit better this time.